packages/core/src/entity/Collection.ts

```diff
@@ -206,14 +206,11 @@ export class Collection<T, O = unknown> extends ArrayCollection<T, O> {
     this.dirty = dirty;
   }
 
-  async init(options?: InitOptions<T>): Promise<this>;
-  async init(populate?: string[], where?: FilterQuery<T>, orderBy?: QueryOrderMap<T>): Promise<this>;
-  async init(populate: string[] | InitOptions<T> = [], where?: FilterQuery<T>, orderBy?: QueryOrderMap<T>): Promise<this> {
-    const options = Utils.isObject<InitOptions<T>>(populate) ? populate : { populate, where, orderBy };
+  async init<P extends string = never>(options?: InitOptions<T, P>): Promise<this> {
```

Comment:
let's add a default value here so we don't need to use optional chaining everywhere (unfortunately it breaks coverage)
```suggestion
  async init<P extends string = never>(options: InitOptions<T, P> = {}): Promise<this> {
```

(and please remove the optional chaining usage underneath)